common/src/lib.rs

```diff
@@ -176,7 +176,14 @@ impl BlockHeader {
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct MerkleProof {
-    // TODO
+    pub proof: Vec<MerkleTree>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum MerkleTree {
```

Comment:
Isn't it supposed be `MerkleProofEntry`?
`MerkleTree` is more like `Vec<Vec<Hash256>>`

Reply:
Fixed.